api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java

```diff
@@ -683,19 +732,18 @@ public DbDataset cloneDataSetToWorkspace(
   }
 
   @Override
-  public void deleteDataSet(DbWorkspace dbWorkspace, Long dataSetId) {
-    long dbDataSetId = getDbDataSet(dbWorkspace, dataSetId).get().getDataSetId();
-    dataSetDao.delete(dbDataSetId);
+  public void deleteDataSet(Long dataSetId) {
+    dataSetDao.delete(dataSetId);
   }
 
   @Override
-  public Optional<DbDataset> getDbDataSet(DbWorkspace dbWorkspace, Long dataSetId) {
-    return Optional.of(dataSetDao.findOne(dataSetId));
+  public Optional<DataSet> getDbDataSet(Long dataSetId) {
+    return Optional.of(dataSetMapper.dbModelToClient(dataSetDao.findOne(dataSetId)));
   }
 
   @Override
   public void markDirty(ResourceType resourceType, long resourceId) {
-    List<DbDataset> dbDataSetList = getDataSets(resourceType, resourceId);
+    List<DbDataset> dbDataSetList = getDbDataSets(resourceType, resourceId);
```

Comment:
Again we can create another story: But should this throw an exception if no data sets are found for resourceType and Id ?

Reply:
Good idea, we could also return `false` here instead of an exception -> https://github.com/all-of-us/workbench/blob/9c1af4c461c124a8ced075f168ce055d8ebeb405/api/src/main/java/org/pmiops/workbench/api/DataSetController.java#L381-L381